.taskcluster.yml

```diff
@@ -70,7 +70,12 @@ tasks:
                 owner: ${event.pusher.email}
                 source: ${event.repository.url}
               payload:
-                image: harjgam/web-platform-tests:0.33
+                image:
+                  $if: browser.name == 'webkitgtk_minibrowser'
+                  then:
+                    cl0p3z/web-platform-tests:webkitgtk20190901
```

Comment:
nit: If we have to use a different docker image could we keep the naming convention somewhat similar? Date based naming seems bad because with the way we have to push to dockerhub and then run a PR you often cycle through several variants on the same day…

But ideally we'd use the same docker file for both setups. Is the browser actually being installed into the docker image? If so I don't think we should do that (but I  haven't heard the reasoning). I'd much prefer the docker image to have the necessary settings that allow us to pull in the latest browser version at runtime.

Reply:
> nit: If we have to use a different docker image could we keep the naming convention somewhat similar? Date based naming seems bad because with the way we have to push to dockerhub and then run a PR you often cycle through several variants on the same day…
> 
> But ideally we'd use the same docker file for both setups. 

The reason I have chosen to use Debian instead of Ubuntu [is explained here](https://github.com/web-platform-tests/wpt/pull/18595#issuecomment-537267080).
The idea was to allow using the built-products that the bots at build.webkit.org are producing as nightlies.
But if this is a hurdle, maybe I can re-do this again to use Ubuntu, and then try to deploy a new bot at build.webkit.org producing nightlies targetting Ubuntu LTS so we can have the same image.


> Is the browser actually being installed into the docker image?
> If so I don't think we should do that (but I haven't heard the reasoning). I'd much prefer the docker image to have the necessary settings that allow us to pull in the latest browser version at runtime.

Yes, currently the browser is installed in the image.
And I have realized this is a bad idea because it makes harder to install from different channels (beta/stable/nightly).

My idea was to land this now so we can start testing with the stable channel (it is what is installed on that image) and then re-work it to avoid installing the browser on the image, but instead do it from ./tools/ci/run_tc.py depending on the channel.

Reply:
@clopez you made an attempt to use the same image already, right? Ubuntu has too old system libraries?

Reply:
> @clopez you made an attempt to use the same image already, right? Ubuntu has too old system libraries?

I made an attempt, this is the summary:

* For webkitgtk/stable the system libraries are ok.
* For webkitgtk/nightly some libraries are too old (GStreamer currently), but it happens the same with debian stable. What we do in the bots of build.webkit.org is build newer versions of the required libraries and install them on a third-party directory of the build. The idea I had is to make a tarball with the built-product including this extra libraries, so it can run on the same distro/version that the buildbot runs without issues. 
* For webkitgtk/beta the system libraries are ok.

So, regarding versions of system libraries, using debian stable vs ubuntu LTS the situation is more or less the same. What made me opt for Debian was this possibility of re-using the built-product of the bots at build.webkit.org.

Reply:
@jgraham extra context for you about what I want out of this. WebKitGTK stable releases don't align with Safari stable, but might be nice to have, and start here to prove something working. But it's the bleeding edge WebKitGTK, and specifically ones built from the same WebKit revision as the current Safari TP that is most interesting. With that in hand we can make the most useful comparisons of almost current WebKit trunk on two platforms, and work out what failures are because of platform differences, and which are reliably failing and most worth triaging.

Reply:
OK, so I don't consider the different docker image a blocker in the first instance. It's not ideal, but we can hopefully figure out some way forward on that after this lands. I think that having the browser installed in the image should be fixed before landing this though. My impression is that it's not loads of work; we just need to do something similar to what we're already doing for Chrome. If I"m wrong and we expect a delay of weeks from putting the browser install elsewhere then I'm happy to reconsider.

Reply:
Just pushed a commit here updating the docker image and installing the browser on taskcluster when the test start. Let me know your thoughts about it, thanks